packages/core/src/components/widgets/CheckboxWidget.tsx

```diff
@@ -1,32 +1,46 @@
 import React from "react";
-import PropTypes from "prop-types";
 import { schemaRequiresTrueValue } from "@rjsf/utils";
+import { WidgetProps } from "@rjsf/utils";
 
-function CheckboxWidget(props) {
-  const {
-    schema,
-    id,
-    value,
-    disabled,
-    readonly,
-    label,
-    autofocus,
-    onBlur,
-    onFocus,
-    onChange,
-    registry,
-  } = props;
-
+/** The `CheckBoxWidget` is a widget for rendering boolean properties.
+ *  It is typically used to represent a boolean.
+ *
+ * @param props - The `WidgetProps` for this component
+ */
+function CheckboxWidget<T = any, F = any>({
+  schema,
+  id,
+  value,
+  disabled,
+  readonly,
+  label,
+  autofocus = false,
+  onBlur,
+  onFocus,
+  onChange,
+  registry,
+}: WidgetProps<T, F>) {
   const { DescriptionFieldTemplate } = registry.templates;
   // Because an unchecked checkbox will cause html5 validation to fail, only add
   // the "required" attribute if the field value must be "true", due to the
   // "const" or "enum" keywords
   const required = schemaRequiresTrueValue(schema);
 
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    return onChange(event.target.checked);
+  };
```

Comment:
While you don't have to make the change unless you want to, there is an even shorter notation for this pattern of fat-arrow function that looks like:
```
const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => onChange(event.target.checked);
```

NOTE: IF you do make the change, so it consistently throughout the PR please.